mlflow/projects/utils.py

```diff
@@ -200,8 +200,11 @@ def _fetch_git_repo(uri, version, dst_dir):
             )
     else:
         origin.fetch(depth=GIT_FETCH_DEPTH)
-        repo.create_head("master", origin.refs.master)
-        repo.heads.master.checkout()
+        ref = origin.refs[0]
```

Comment:
I test on my repo `git@github.com:WeichenXu123/mlflow.git`
but the `origin.refs[0]` I got `<git.RemoteReference "refs/remotes/origin/0.9">`, which is not master or main branch.

Reply:
It looks like `origin.fetch(depth=GIT_FETCH_DEPTH)` fetches all branches.